lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -204,4 +204,14 @@ def delete(volume)
       volume
     end
   end
+
+  def availability_zone
```

Comment:
But they could set their region to `eu-west-1` in their `with_driver` block and write their availability_zone as `us-east-2a`.  With the current code, it would put their EBS volume in `eu-west-1a`.

If you do something like the following:
```ruby
if /^#{region}/ =~ az
  Chef::Log.warn("availability_zone attribute should only be set to the letter designation. Attempting to use '#{az[-1]}' to correct the issue.")
elsif az.length == 1
  az = "#{region}#{az[-1]}"
end
```

Then you ensure that (if provided) the full availability zone matches the current region.  If it doesn't match, amazon will get angry that they're trying to put it in an availability zone that doesn't exist in the region.

What validation are you talking about?  I couldn't find any in the provider or the resource.